src/shared/services/ipc/index.js

```diff
@@ -0,0 +1,61 @@
+import { uniqueId } from 'lodash';
+
+export const responseMessageHandler = callback => async (event, { id, payload } = {}) => {
+  try {
+    return event.sender.send(
+      `${id}/result`,
+      {
+        id,
+        type: 'resolve',
+        payload: await callback(...payload),
+      },
+    );
+  } catch (err) {
+    return event.sender.send(
+      `${id}/result`,
+      {
+        id,
+        type: 'reject',
+        payload: err,
+      },
+    );
+  }
+};
+
+export const makeIpcPromiseWrapper = (ipc) => {
+  const send = async (channel, ...args) => {
+    // Forge unique id.
+    const id = uniqueId(`ipc_${channel}_`);
+
+    // Send an ipc request.
+    return new Promise((resolve, reject) => {
+      // Send request.
+      ipc.send(
+        channel,
+        {
+          id,
+          payload: args,
+        },
+      );
+
+      // wait for resolution or rejection
+      ipc.once(`${id}/result`, (event, data) => {
+        if (data.type === 'resolve') {
+          resolve(data.payload);
+        } else {
+          reject(data.payload);
+        }
+      });
+    });
+  };
+
+  const on = (channel, callback) => ipc.on(channel, responseMessageHandler(callback));
```

Comment:
does that handle errors ?

Reply:
It’s the job of the other end of the ipc channel to catch the error and send it back, otherwise we would need to add a timeout (ipc channels are only 1-way so we can’t know if it crashed if the error is not sent back via channel/result).